exercises/kindergarten-garden/test/Tests.hs

```diff
@@ -1,54 +1,73 @@
-    [Clover, Grass, Radishes, Grass] @=? l "Eve"
-    [Grass, Clover, Violets, Clover] @=? l "Fred"
-    [Clover, Grass, Grass, Clover] @=? l "Ginny"
-    [Violets, Radishes, Radishes, Violets] @=? l "Harriet"
-    [Grass, Clover, Violets, Clover] @=? l "Ileana"
-    [Violets, Clover, Violets, Grass] @=? l "Joseph"
-    [Grass, Clover, Clover, Grass] @=? l "Kincaid"
-    [Grass, Violets, Clover, Violets] @=? l "Larry"
-  , testCase "surprise garden" $ do
-    let g = garden [ "Samantha", "Patricia"
-                   , "Xander", "Roger"
-                   ] "VCRRGVRG\nRVGCCGCV"
-        l = flip lookupPlants g
-    [Violets, Clover, Radishes, Violets] @=? l "Patricia"
-    [Radishes, Radishes, Grass, Clover] @=? l "Roger"
-    [Grass, Violets, Clover, Grass] @=? l "Samantha"
-    [Radishes, Grass, Clover, Violets] @=? l "Xander"
-  ]
+main = hspecWith defaultConfig {configFastFail = True} specs
+
+specs :: Spec
+specs = describe "kindergarten-garden" $ do
+
+    -- As of 2016-08-02, there was no reference file
+    -- for the test cases in `exercism/x-common`.
+
+    it "alice tests" $ do
+
+      let alicePlants = lookupPlants "Alice" . defaultGarden
+
+      alicePlants "RC\nGG" `shouldBe` [ Radishes, Clover, Grass   , Grass  ]
+      alicePlants "VC\nRC" `shouldBe` [ Violets , Clover, Radishes, Clover ]
+
+    it "small garden" $ do
+
+      let plants s  = lookupPlants s $ defaultGarden plantList
+          plantList = "VVCG\nVVRC"
+
+      plants "Bob" `shouldBe` [ Clover, Grass, Radishes, Clover ]
+
+    it "medium garden" $ do
+
+      let plants s  = lookupPlants s $ defaultGarden plantList
+          plantList = "VVCCGG\nVVCCGG"
+
+      plants "Bob"     `shouldBe` [ Clover, Clover, Clover, Clover ]
+      plants "Charlie" `shouldBe` [ Grass , Grass , Grass , Grass  ]
+
+    it "full garden" $ do
+
+      let plants s  = lookupPlants s $ defaultGarden plantList
+          plantList = "VRCGVVRVCGGCCGVRGCVCGCGV\nVRCCCGCRRGVCGCRVVCVGCGCV"
+
+      plants "Alice"   `shouldBe` [ Violets , Radishes, Violets , Radishes ]
+      plants "Bob"     `shouldBe` [ Clover  , Grass   , Clover  , Clover   ]
+      plants "Charlie" `shouldBe` [ Violets , Violets , Clover  , Grass    ]
+      plants "David"   `shouldBe` [ Radishes, Violets , Clover  , Radishes ]
+      plants "Eve"     `shouldBe` [ Clover  , Grass   , Radishes, Grass    ]
+      plants "Fred"    `shouldBe` [ Grass   , Clover  , Violets , Clover   ]
+      plants "Ginny"   `shouldBe` [ Clover  , Grass   , Grass   , Clover   ]
+      plants "Harriet" `shouldBe` [ Violets , Radishes, Radishes, Violets  ]
+      plants "Ileana"  `shouldBe` [ Grass   , Clover  , Violets , Clover   ]
+      plants "Joseph"  `shouldBe` [ Violets , Clover  , Violets , Grass    ]
+      plants "Kincaid" `shouldBe` [ Grass   , Clover  , Clover  , Grass    ]
+      plants "Larry"   `shouldBe` [ Grass   , Violets , Clover  , Violets  ]
+
+    it  "surprise garden" $ do
```

Comment:
I wonder what's so surprising about this garden. Is it just the different student names? Perhaps we'll not know the answer, no explanation was given in https://github.com/exercism/exercism.io/pull/595